rotkehlchen/chain/manager.py

```diff
@@ -1218,185 +867,163 @@ def remove_blockchain_accounts(
                 f'accounts {",".join(unknown_accounts)}',
             )
 
-        # If no blockchain query has happened before then we need to query the relevant
-        # chain to populate the self.balances mapping. But query has to happen after
-        # account removal so as not to query unneeded accounts
-        balances_queried_before = True
-        if not self.balances.is_queried(blockchain):
-            balances_queried_before = False
-
         self.modify_blockchain_accounts(
             blockchain=blockchain,
             accounts=accounts,
             append_or_remove='remove',
-            add_or_sub=operator.sub,
         )
 
-        if not balances_queried_before:
-            self.query_balances(blockchain, ignore_cache=True)
+    def modify_dot_ksm_accounts(
+            self,
+            accounts: Union[List[PolkadotAddress], List[KusamaAddress]],
+            blockchain: Literal[SupportedBlockchain.POLKADOT, SupportedBlockchain.KUSAMA],
+            append_or_remove: Literal['append', 'remove'],
+    ) -> None:
+        """May raise:
+        - RemoteError if we couldn't connect to any node"""
+        if blockchain == SupportedBlockchain.POLKADOT:
+            connection_timeout = POLKADOT_NODE_CONNECTION_TIMEOUT
+        else:  # Kusama
+            connection_timeout = KUSAMA_NODE_CONNECTION_TIMEOUT
```

Comment:
They are the same anyway.

```suggestion
        connection_timeout = SUBSTRATE_NODE_CONNECTION_TIMEOUT
```